### PR TITLE
Update changelog

### LINKED ISSUE #231: Update changelog for 1.2.0
Here's what I was able to dig up about 1.1.0 -> 1.2.0

**related to the schema**

1. switched to JSONschema
1. `issue`, `number`, `version`, `post-code`, `section` are more lenient now with type union `str|number`
1. `version`, `date-released` no longer required elements of a minimal CFF
1. list of valid license SPDX codes updated to version 2021-05-14
1. `language` more lenient (for performance)
1. `identifiers` have gained an optional key `description`
1. added regex validation for identifiers of type `swh`
1. `description`s and `examples` added to schema
1. schema has more checks for empty strings or empty arrays (e.g. `authors`, `abstract`, `keywords`)
1. root document now has `type:` `software` or `dataset`
1. regex for `url` simplified for easier maintenance
1. url regex now also allows for `sftp`
1. regex for `isbn` simplified and changed for easier maintenance
1. introduced `preferred-citation`
1. dates are now (only) strings of `format` `date` & `pattern` YYYY-MM-DD
1. `references.term` was added

> Ignore below:
>
> **related to the rest of the repo**
> 
> 1. added test cases for 1.2.0
> 1. removed the duplication from the testing setup, all CITATION.cff files are now just fixtures with which one test is run
> 1. removed any travis related things, switched to github actions
> 1. added contributors
> 1. expanded the contributing guide
> 2. Make README more concise
> 3. Add the CFF Schema Guide

_Originally posted by @jspaaks in https://github.com/citation-file-format/citation-file-format/issues/222#issuecomment-866929216_

### PR DESCRIPTION
**Pull request by sdruskat**
**Related issues**

Fix #231 

**Describe the changes made in this pull request**

- Update changelog for 1.2.0

**Instructions to review the pull request**

Look at changelog, see if you find anything that's missing

### CODE REVIEW

**Comment by sdruskat** [comment]
Think this is good to go now.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,33 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
```
Comment (jspaaks) [inline CHANGELOG.md:9]: I think there should be a caption 1.2.0 in between Unreleased and 1.1.10 right?

**Review by jspaaks — CHANGES_REQUESTED** [review]
Looks good as far as I can tell. I scanned throguh https://github.com/citation-file-format/citation-file-format/compare/cdb00ac29f2458321edf392f049b7ec3e3c4668f..HEAD didnt see anything that wasnt covered by the notes.

Just need to add the title for the release with a link

File: CHANGELOG.md (hunk 1 above)
Comment (sdruskat) [reply CHANGELOG.md:9]: Ouch, sorry. I'm used to automating this with a Maven plugin for another project I'm working on. I'll add the header.

**Comment by sdruskat** [comment]
> Looks good as far as I can tell. I scanned throguh https://github.com/citation-file-format/citation-file-format/compare/cdb00ac29f2458321edf392f049b7ec3e3c4668f..HEAD (the state of the repo since I made [#222 (comment)](https://github.com/citation-file-format/citation-file-format/pull/222#issuecomment-866929216)) --didnt see anything that wasnt covered by the notes.
> 
> Just need to add the title for the release with a link

Thanks for the review and catching this! Better now?

**Comment by sdruskat** [comment]
(Failing link is `1.2.0` to `HEAD` diff, fair enough...

**Review by jspaaks — APPROVED** [review]
:tada: :tada: :tada: :tada: :tada:

**Pull request merged**